app/components/post_draft/draft_input/draft_input.test.tsx

```diff
@@ -0,0 +1,166 @@
+import NetworkManager from '@managers/network_manager';
+import {openAsBottomSheet} from '@screens/navigation';
+import {renderWithEverything} from '@test/intl-test-helper';
+import TestHelper from '@test/test_helper';
+import {persistentNotificationsConfirmation} from '@utils/post';
+
+import DraftInput from './draft_input';
+
+import type ServerDataOperator from '@database/operator/server_data_operator';
+import type {Database} from '@nozbe/watermelondb';
+
+const SERVER_URL = 'https://appv1.mattermost.com';
+
+// this is needed to when using the useServerUrl hook
+jest.mock('@context/server', () => ({
+    useServerUrl: jest.fn(() => SERVER_URL),
+}));
+
+jest.mock('@screens/navigation', () => ({
+    openAsBottomSheet: jest.fn(),
+}));
+
+jest.mock('@utils/post', () => ({
+    persistentNotificationsConfirmation: jest.fn(),
+}));
+
+describe('DraftInput', () => {
+    const baseProps = {
+        testID: 'draft_input',
+        channelId: 'channelId',
+        channelType: 'O' as ChannelType,
+        currentUserId: 'currentUserId',
+        postPriority: {priority: ''} as PostPriority,
+        updatePostPriority: jest.fn(),
+        persistentNotificationInterval: 0,
+        persistentNotificationMaxRecipients: 0,
+        updateCursorPosition: jest.fn(),
+        cursorPosition: 0,
+        sendMessage: jest.fn(),
+        canSend: true,
+        maxMessageLength: 4000,
+        files: [],
+        value: '',
+        uploadFileError: null,
+        updateValue: jest.fn(),
+        addFiles: jest.fn(),
+        updatePostInputTop: jest.fn(),
+        setIsFocused: jest.fn(),
+        scheduledPostsEnabled: true,
+    };
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    let database: Database;
+    let operator: ServerDataOperator;
+
+    beforeEach(async () => {
+        const server = await TestHelper.setupServerDatabase(SERVER_URL);
+        database = server.database;
+        operator = server.operator;
+    });
+
+    afterEach(async () => {
+        await TestHelper.tearDown();
+        NetworkManager.invalidateClient(SERVER_URL);
+    });
+
+    describe('Rendering', () => {
+        it('renders base components', async () => {
+            const {getByTestId} = renderWithEverything(<DraftInput {...baseProps}/>, {database});
+            expect(getByTestId('draft_input')).toBeTruthy();
+            expect(getByTestId('draft_input.post.input')).toBeTruthy();
+            expect(getByTestId('draft_input.quick_actions')).toBeTruthy();
+            expect(getByTestId('draft_input.send_action.send.button')).toBeTruthy();
+        });
+
+        it('shows upload error when present', () => {
+            const props = {...baseProps, uploadFileError: 'Error message'};
+            const {getByText} = renderWithEverything(<DraftInput {...props}/>, {database});
+            expect(getByText('Error message')).toBeTruthy();
+        });
+    });
+
+    describe('Message Actions', () => {
```

Comment:
There are a couple of details you are not testing. Mainly that you are passing the right action to the popups (i.e. `openAsBottomSheet` and `persistentNotificationsConfirmation`).

What could be done to test that is the following:
- Mock the implementation of `openAsBottomSheet` (and similar for `persistentNotificationsConfirmation`) like this:
```
jest.mocked(openAsBottomSheet).mockImplementation(({props: {onSchedule}}) => onSchedule(SCHEDULING_INFO))
```
- Verify send message is called
- Verify that if you had post priority, you still get the warning

This way you verify that if the bottom sheet (or the verification for priority messages) does what you expect them to do, your code does things right (in other words... you are passing the right functions into the right places).